.github/workflows/e2e-tests.yml

```
@@ -0,0 +1,61 @@
+# This workflow builds the connector project and executes E2E tests using Gradle. It is triggered by pull request
+# openings and synchronization, ensuring that the changes introduced by the PR are validated and do not break the
+# connector's functionality.
+
+name: Connector E2E Tests
+
+on:
+  pull_request:
+    branches:
+      - "develop"
+    types:
+      - opened
+      - synchronize
```

This will only trigger the CI if the branch is develop, please remove

```suggestion
on:
  pull_request:
    types: [opened, reopened, synchronize]
```
```suggestion
on:
  pull_request:
    branches:
      - "develop"
    types:
      - opened
      - synchronize
```

---

> This will only trigger the CI if the branch is develop, please remove
> 
This triggers the workflow if the target branch is develop, which I believe is the intended behavior. Each PR targeting develop will execute this workflow. For other target branches (if we omit branches: - "develop"), I assume there will be a different workflow configuration. 
Also, I will add the reopened type.